### PR TITLE
Prepare to phpunit10

### PR DESCRIPTION
**Pull request by Goodmain**
Refs #99

### CODE REVIEW

File: .github/workflows/run-tests-with-coverage.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     steps:
       - uses: shivammathur/setup-php@v2
         with:
-          php-version: '7.4'
+          php-version: '8.1'
```
Comment (DenTray) [inline .github/workflows/run-tests-with-coverage.yml:15]: as we discussed - we need to just implement the compatibility with PhpUnit10, but still use PhpUnit 9 version as an available

File: src/Traits/MockClassTrait.php [hunk 2]
```diff
@@ -31,12 +33,14 @@ public function mockClass(string $class, array $callChain): void
             $mock
                 ->expects($this->exactly($calls->count()))
                 ->method($method)
-                ->withConsecutive(...$calls->map(function ($call) {
-                    return $call['arguments'];
-                })->toArray())
-                ->willReturnOnConsecutiveCalls(...$calls->map(function ($call) {
-                    return $call['result'];
-                })->toArray());
+                ->with(
```
Comment (DenTray) [inline src/Traits/MockClassTrait.php:36]: as discussed, please try to use `willReturnCallback` method to assert input args correctly

**Review by DenTray — CHANGES_REQUESTED** [review]

File: .github/workflows/run-tests-with-coverage.yml (hunk 1 above)
Comment (Goodmain) [reply .github/workflows/run-tests-with-coverage.yml]: @DenTray Done

File: src/Traits/MockClassTrait.php (hunk 2 above)
Comment (Goodmain) [reply src/Traits/MockClassTrait.php]: @DenTray Done

File: tests/HttpRequestServiceTest.php [hunk 3]
```diff
@@ -35,19 +34,21 @@ public function setUp(): void
...
+        $this->mockClass(HttpRequestService::class, [
+            [
+                'method' => 'sendRequest',
+                'arguments' => [
+                    'get', 'https://some.url.com', [
+                        'some_key' => 'some_value'
+                    ], [
+                        'some_header_name' => 'some_header_value'
+                    ]
+                ],
```
Comment (DenTray) [inline tests/HttpRequestServiceTest.php:46]:
```suggestion
                'arguments' => [
                    'get',
                    'https://some.url.com',
                    ['some_key' => 'some_value'],
                    ['some_header_name' => 'some_header_value']
                ],
```

**Review by DenTray — CHANGES_REQUESTED** [review]

**Review by DenTray — APPROVED** [review]

**Pull request merged**